test/functional/test_framework/ltc_util.py

```diff
@@ -56,46 +61,86 @@ def make_utxo(node, amount, confirmed=True, scriptPubKey=DUMMY_P2WPKH_SCRIPT):
 
     return COutPoint(int(txid, 16), 0)
 
-def setup_mweb_chain(node):
+
+"""Generates all pre-MWEB blocks, pegs 1 coin into the MWEB,
+then mines the first MWEB block which includes that pegin.
+
+mining_node - The node to use to generate blocks
+"""
+def setup_mweb_chain(mining_node):
     # Create all pre-MWEB blocks
-    node.generate(431)
+    mining_node.generate(FIRST_MWEB_HEIGHT - 1)
 
     # Pegin some coins
-    node.sendtoaddress(node.getnewaddress(address_type='mweb'), 1)
+    mining_node.sendtoaddress(mining_node.getnewaddress(address_type='mweb'), 1)
 
     # Create some blocks - activate MWEB
-    node.generate(1)
+    mining_node.generate(1)
 
-def get_hog_addr_txout(node):
-    best_block = node.getblock(node.getbestblockhash(), 2)
 
-    hogex_tx = best_block['tx'][-1] # TODO: Should validate that the tx is marked as a hogex tx
-    hog_addr = hogex_tx['vout'][0]
+"""Retrieves the HogEx transaction for the block.
 
-    return CTxOut(int(hog_addr['value'] * COIN), hog_addr['scriptPubKey'])
+node - The node to use to lookup the transaction.
+block_hash - The block to retrieve the HogEx for. If not provided, the best block hash will be used.
 
-def get_mweb_header_tip(node):
-    best_block = node.getblock(node.getbestblockhash(), 2)
+Returns the HogEx as a 'CTransaction'
+"""
+def get_hogex_tx(node, block_hash = None):
+    block_hash = block_hash or node.getbestblockhash()
+    hogex_tx_id = node.getblock(block_hash)['tx'][-1]
+    hogex_tx = FromHex(CTransaction(), node.getrawtransaction(hogex_tx_id, False, block_hash)) # TODO: Should validate that the tx is marked as a hogex tx
+    hogex_tx.rehash()
+    return hogex_tx
+    
+
+"""Retrieves the HogAddr for a block.
+
+node - The node to use to lookup the HogAddr.
+block_hash - The block to retrieve the HogAddr for. If not provided, the best block hash will be used.
+
+Returns the HogAddr as a 'CTxOut'
+"""
+def get_hog_addr_txout(node, block_hash = None):
+    return get_hogex_tx(node, block_hash).vout[0]
```

Comment:
The old code instantiated a CTxOut, was that unnecessary?

Reply:
The old code only had access to the JSON response, not a parsed CTransaction, so we had to built a CTxOut from the relevant fields. The new code parses the whole JSON into a CTransaction, which then contains an actual CTxOut already, so we no longer need to build it ourselves.